MatrixKit/Models/Account/MXKAccountManager.m

```diff
@@ -468,12 +468,13 @@ - (void)setPushDeviceToken:(NSData *)pushDeviceToken withPushOptions:(NSDictiona
             // turn off the Push flag for all accounts if any
             for (MXKAccount *account in mxAccounts)
             {
-                [account enablePushKitNotifications:NO success:nil failure:nil];
+                [account enablePushKitNotifications:NO success:^{
+                    //  make sure pusher really removed before losing token.
+                    [[NSUserDefaults standardUserDefaults] removeObjectForKey:@"pushDeviceToken"];
+                    [[NSUserDefaults standardUserDefaults] removeObjectForKey:@"pushOptions"];
+                } failure:nil];
```

Comment:
We can add a log here for rageshakes investigation.

Reply:
It’s already logged at https://github.com/matrix-org/matrix-ios-kit/blob/1d11c01e97fdf89b5ed5ee457fce9a6552c3ef9a/MatrixKit/Models/Account/MXKAccount.m#L530